instance/tasks.py

```diff
@@ -78,3 +77,16 @@ def watch_pr():
             if created:
                 logger.info('New PR found, creating sandbox: %s', pr)
                 provision_instance(instance.pk)
+
+@periodic_task(crontab(minute='*/1'))
+def watch_branch_changes():
+    """
+    Re-provision an existing instance whenever a branch has been updated.
+    TODO: what we really want to do is to create a new instance and delete the
+    previous one once the new one is ready.
+    """
+    for instance in OpenEdXInstance.objects.all():
+        commit_id = instance.commit_id
+        new_commit_id = instance.get_branch_tip_commit_id()
+        if new_commit_id != commit_id:
+            instance.provision()
```

Comment:
Yup, good approach. I would however add a flag on a per-instance basis to activate this, with a default to false - to ensure that existing sandboxes for which people have done changes manually don't get automatically reprovisioned unless it's explicitly activated. And for sandboxes, the field could be set from the PR description - for an example see:

* https://github.com/open-craft/opencraft/pull/11/files#diff-a3f79e2fa4816b6091c587473626caf2R90
* https://github.com/open-craft/opencraft/pull/11/files#diff-04c6e90faac2675aa89e2176d2eec7d8R235